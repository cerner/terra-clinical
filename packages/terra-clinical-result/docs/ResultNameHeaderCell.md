# Result Name Header Cell

Result-Name-Header-Cell is a component of terra-clinical-result that creates a standardized view for the Result Name and Unit of Measure in the first column (row header) for each row within a flowsheet.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-clinical-result`


<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
<!-- AUTO-GENERATED-CONTENT:END -->


## Component Features
