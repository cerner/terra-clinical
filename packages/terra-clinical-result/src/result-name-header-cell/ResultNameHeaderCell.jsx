import React from 'react';
import PropTypes from 'prop-types';
import IconCalculator from 'terra-icon/lib/icon/IconCalculator';
import classNames from 'classnames/bind';
import styles from './ResultNameHeaderCell.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Content to be displayed for the Result Name row label.
   */
  resultName: PropTypes.string.isRequired,
  /**
   * Content to be displayed as the supporting Unit of Measure row label.
   */
  unit: PropTypes.string,
  /**
   * The padding styling to apply to the Result Name row header cell.
   * One of `'none'`, `'compact'`, or `'standard'`.
   */
  paddingStyle: PropTypes.oneOf(['none', 'compact', 'standard']),
  /**
   * Adds additional icon at beginning of row before the ResultName, indicating special information about results or cells.
   * One of `'none'` or `'calculated'`.
   */
  typeIndicator: PropTypes.oneOf(['none', 'calculated']),
};

const defaultProps = {
  paddingStyle: 'compact',
  typeIndicator: 'none',
};

const resultRowIndicators = {
  none: null,
  calculated: <IconCalculator className={cx('icon-rowindicator')} />,
};

const ResultNameHeaderCell = (props) => {
  const {
    resultName,
    unit,
    paddingStyle,
    typeIndicator,
    ...customProps
  } = props;

  const nameHeaderCellClassnames = cx([
    'clinical-result-name-header-cell',
    { 'padding-standard': paddingStyle === 'standard' },
    { 'padding-compact': paddingStyle === 'compact' },
  ]);

  return (
    <div
      {...customProps}
      className={customProps.className ? `${nameHeaderCellClassnames} ${customProps.className}` : nameHeaderCellClassnames}
    >
      <div className={cx('name')}>
        {resultRowIndicators[typeIndicator.toLowerCase()]}
        {resultName}
      </div>
      {unit && <div className={cx('unit')}>{unit}</div> }
    </div>
  );
};

ResultNameHeaderCell.propTypes = propTypes;
ResultNameHeaderCell.defaultProps = defaultProps;

export default ResultNameHeaderCell;
