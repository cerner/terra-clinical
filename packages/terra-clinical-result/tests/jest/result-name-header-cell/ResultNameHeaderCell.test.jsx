import React from 'react';
import ResultNameHeaderCell from '../../../src/result-name-header-cell/ResultNameHeaderCell';

// Snapshot Tests
describe('ResultNameHeaderCell', () => {
  it('should render a result name', () => {
    const cell = shallow(<ResultNameHeaderCell resultName="Name" />);
    expect(cell).toMatchSnapshot();
  });

  it('should render a unit', () => {
    const cell = shallow(<ResultNameHeaderCell resultName="Name" unit="Unit" />);
    expect(cell).toMatchSnapshot();
  });

  describe('paddingStyle -', () => {
    it('should render with none', () => {
      const cell = shallow(<ResultNameHeaderCell resultName="Name" paddingStyle="none" />);
      expect(cell).toMatchSnapshot();
    });

    it('should render with standard', () => {
      const cell = shallow(<ResultNameHeaderCell resultName="Name" paddingStyle="standard" />);
      expect(cell).toMatchSnapshot();
    });

    it('should render with compact', () => {
      const cell = shallow(<ResultNameHeaderCell resultName="Name" paddingStyle="compact" />);
      expect(cell).toMatchSnapshot();
    });
  });

  describe('typeIndicator -', () => {
    it('should render with none', () => {
      const cell = shallow(<ResultNameHeaderCell resultName="Name" typeIndicator="none" />);
      expect(cell).toMatchSnapshot();
    });

    it('should render with calculated', () => {
      const cell = shallow(<ResultNameHeaderCell resultName="Name" typeIndicator="calculated" />);
      expect(cell).toMatchSnapshot();
    });
  });
});
