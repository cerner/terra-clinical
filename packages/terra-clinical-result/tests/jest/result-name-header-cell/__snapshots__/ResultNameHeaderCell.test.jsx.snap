// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultNameHeaderCell paddingStyle - should render with compact 1`] = `
<div
  className="clinical-result-name-header-cell padding-compact"
>
  <div
    className="name"
  >
    Name
  </div>
</div>
`;

exports[`ResultNameHeaderCell paddingStyle - should render with none 1`] = `
<div
  className="clinical-result-name-header-cell"
>
  <div
    className="name"
  >
    Name
  </div>
</div>
`;

exports[`ResultNameHeaderCell paddingStyle - should render with standard 1`] = `
<div
  className="clinical-result-name-header-cell padding-standard"
>
  <div
    className="name"
  >
    Name
  </div>
</div>
`;

exports[`ResultNameHeaderCell should render a result name 1`] = `
<div
  className="clinical-result-name-header-cell padding-compact"
>
  <div
    className="name"
  >
    Name
  </div>
</div>
`;

exports[`ResultNameHeaderCell should render a unit 1`] = `
<div
  className="clinical-result-name-header-cell padding-compact"
>
  <div
    className="name"
  >
    Name
  </div>
  <div
    className="unit"
  >
    Unit
  </div>
</div>
`;

exports[`ResultNameHeaderCell typeIndicator - should render with calculated 1`] = `
<div
  className="clinical-result-name-header-cell padding-compact"
>
  <div
    className="name"
  >
    <IconCalculator
      className="icon-rowindicator"
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    Name
  </div>
</div>
`;

exports[`ResultNameHeaderCell typeIndicator - should render with none 1`] = `
<div
  className="clinical-result-name-header-cell padding-compact"
>
  <div
    className="name"
  >
    Name
  </div>
</div>
`;
